test/utils/asserts.hpp

```diff
@@ -54,3 +55,18 @@ namespace fizzy
 {
 std::ostream& operator<<(std::ostream& os, ExecutionResult);
 }
+
+namespace fizzy::test
+{
+inline uint32_t as_uint32(fizzy::Value value)
+{
+    EXPECT_EQ(value.i64 & 0xffffffff00000000ULL, 0);
+    return static_cast<uint32_t>(value.i64);
+}
+
+inline int32_t as_int32(fizzy::Value value)
+{
+    EXPECT_EQ(value.i64 & 0xffffffff00000000ULL, 0);
```

Comment:
```suggestion
    EXPECT_EQ(value.i64 & 0xffffffff00000000, 0);
```
Yes, this looks good now. The suggestion should work too I think.

Alternatively, we may consider to add the assert to `Value::as<uint32_t>()`.

Reply:
I had it without the postfix, but wanted to be sure. Perhaps not needed in C++.

Reply:
> Alternatively, we may consider to add the assert to Value::as<uint32_t>().

Maybe we should do that :wink:

Reply:
Added it, but we already have a trap with the unit tests:
```
[ RUN      ] end_to_end.milestone2
Assertion failed: ((i64 & 0xffffffff00000000) == 0), function as, file /Users/alex/Projects/fizzy/lib/fizzy/value.hpp, line 59.
Abort trap: 6
```

Reply:
Created an issue to track this.